tests/rptest/scale_tests/many_topics_test.py

```diff
@@ -52,48 +55,70 @@ class ManyTopicsTest(RedpandaTest):
     # Max time to wait for the cluster to be healthy once more.
     HEALTHY_WAIT_SECONDS = 20 * 60
 
-    # Max topic count supported by single client-swarm node
-    MAX_SWARM_NODE_TOPIC_COUNT = 15_000
+    # Max client count supported by single client-swarm node
+    # (assuming a m6id.xlarge or equiv node).
+    MAX_SWARM_NODE_CLIENT_COUNT = 1_000
```

Comment:
Seems small?

In any case I don't think we should hardcode a value which depends on the instance type. Maybe the quickest fix is just to `assert` the memory/core count is as expected, with a message like "swarm node max needs to be updated" or something? I think `ScaleParameters` have the necessary hooks to figure that stuff out.

Reply:
@ballard26 unresolved?

Reply:
Adjusted the limit to be relative to the amount of memory per client and the max clients per core as discussed.